### PR TITLE
schedule: a schedule framework

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 

- implement a schedule framework
- implement a table-number-based scheduler
- add some tests

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

test rebanlance manually.

### CODE REVIEW

File: cdc/model/owner.go [hunk 1]
```diff
@@ -100,13 +99,31 @@ func (o *TableOperation) Clone() *TableOperation {
 	return &clone
 }
 
+// TaskWorkload records the workloads of a task
+type TaskWorkload map[int64]uint64
```
Comment (overvenus) [inline cdc/model/owner.go:103]: What do int64 and uint64 mean?

Comment (overvenus) [reply cdc/model/owner.go:103]: Can we use a struct instead of uint64? Struct is more extendable.

**Comment by leoppro** [comment]
@overvenus  PTAL again

File: cdc/model/owner.go [hunk 2]
```diff
@@ -100,13 +99,32 @@ func (o *TableOperation) Clone() *TableOperation {
 	return &clone
 }
 
+// TaskWorkload records the workloads of a task
+// the value of the struct is the workload
+type TaskWorkload map[TableID]uint64
```
Comment (overvenus) [inline cdc/model/owner.go:104]: What about this? So we can add more fields if needed.
```
type TaskWorkload map[TableID]struct { something uint64 }
```

Comment (leoppro) [reply cdc/model/owner.go:104]:
```
package main

import (
	"fmt"
)

type TaskWorkload map[int]struct{ workload uint64 }

func main() {
	a := make(TaskWorkload)

	// we have to insert value like this ....
	a[1] = struct{ workload uint64 }{workload: 1}
	b := a[1].workload
	fmt.Println(b)
}
```

**Comment by leoppro** [comment]
@overvenus  PTAL again

**Pull request merged**